service/config_provider.go

```diff
@@ -163,11 +187,20 @@ func (cm *configProvider) Shutdown(ctx context.Context) error {
 	return multierr.Combine(cm.closeIfNeeded(ctx), mergeShutdown(ctx, cm.configMapProviders))
 }
 
-func mergeRetrieve(ctx context.Context, onChange func(*configmapprovider.ChangeEvent), providers []configmapprovider.Provider) (configmapprovider.Retrieved, error) {
+func mergeRetrieve(ctx context.Context, onChange func(*configmapprovider.ChangeEvent), selectors []string, providers map[string]configmapprovider.Provider) (configmapprovider.Retrieved, error) {
```

Comment:
We have `map[string]configmapprovider.Provider` in many places but it is not clear what `string` means here. Can we define a type alias (either for the key or for the map) to make the type of the key clear?

Reply:
Reduced the usage to just one place, added comment about the meaning. Please reopen if anything is missing.